backend/test/integration/langchain.test.ts

```diff
@@ -11,23 +22,19 @@ import {
 	promptEvalContextTemplate,
 	promptEvalPrompt,
 } from '@src/promptTemplates';
-import { RetrievalQAChain } from 'langchain/chains';
-import { PromptTemplate } from 'langchain/prompts';
-
-import { ChatOpenAI } from 'langchain/chat_models/openai';
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
-const mockCall = jest.fn();
+const mockCall = jest.fn<any>();
 const mockRetrievalQAChain = {
 	call: mockCall,
 };
 const mockPromptEvalChain = {
 	call: mockCall,
 };
-const mockFromLLM = jest.fn();
-const mockFromTemplate = jest.fn();
+const mockFromLLM = jest.fn<any>();
+const mockFromTemplate = jest.fn<typeof PromptTemplate.fromTemplate>();
```

Comment:
Is this correct? The type args of jest.fn, where given, appear to be the return value and then the args of the function. Therefore I would expect this either to be an untyped `jest.fn()`, or

```
jest.fn<
  ReturnType<typeof PromptTemplate.fromTemplate>,
  Parameters<typeof PromptTemplate.fromTemplate>
>();
```

Or are the types provided by @jest/globals different to those provided by @types/jest?

Reply:
I tried your suggestion and it gets this error:

![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/b5b96e2f-54cf-4106-bdcf-d36cbedcfa26)

If we look at [the docs](![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/bd20a5db-09ed-43a1-8b5f-2958d09a4584)), it looks as though the type arg is supposed to be the type of the method we are mocking, which is how I did the other non-generic type args.

Relevant examples from the doc:
![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/255999ea-8af7-424c-8015-f8d6ee34add1)
![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/2e704774-d849-4b87-897c-4477470fb2fb)
![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/fce013c7-3e08-4433-8071-0db2d6d225a1)

Reply:
Ok, so `@types/jest` is in fact wrong! Good to know.